### ISSUE TITLE
ft.dfs with cutoff dataframe with different order of columns fails

### ISSUE THREAD

**Issue by sebrahimi1988** [FeatureLabs/featuretools#487]
I was creating features for this entity set:

![Screenshot 2019-04-10 at 17 29 59](https://user-images.githubusercontent.com/25546598/55896486-4f0df100-5bb6-11e9-9050-2f43bda2babc.png)

 with the following cutoff parameter:

![Screenshot 2019-04-10 at 17 25 24](https://user-images.githubusercontent.com/25546598/55896281-ddce3e00-5bb5-11e9-80bb-e9640e835a36.png)

and this command:
```
feature_matrix_2018, feature_defs = ft.dfs(entityset=es,
                                           target_entity='users',
                                           cutoff_time=ct,
                                           cutoff_time_in_index=False)
```
I was getting this error: 

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/miniconda3/lib/python3.7/site-packages/pandas/core/arrays/categorical.py in __init__(self, values, categories, ordered, dtype, fastpath)
    357             try:
--> 358                 codes, categories = factorize(values, sort=True)
    359             except TypeError:

~/miniconda3/lib/python3.7/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    187                     kwargs[new_arg_name] = new_arg_value
--> 188             return func(*args, **kwargs)
    189         return wrapper

~/miniconda3/lib/python3.7/site-packages/pandas/core/algorithms.py in factorize(values, sort, order, na_sentinel, size_hint)
    612                                            size_hint=size_hint,
--> 613                                            na_value=na_value)
    614 

~/miniconda3/lib/python3.7/site-packages/pandas/core/algorithms.py in _factorize_array(values, na_sentinel, size_hint, na_value)
    459     uniques, labels = table.factorize(values, na_sentinel=na_sentinel,
--> 460                                       na_value=na_value)
    461 

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.Int64HashTable.factorize()

ValueError: Buffer has wrong number of dimensions (expected 1, got 2)

During handling of the above exception, another exception occurred:

NotImplementedError                       Traceback (most recent call last)
<ipython-input-67-36cde9f8fcf4> in <module>
      2                                            target_entity='users',
      3                                            cutoff_time=ct,
----> 4                                            cutoff_time_in_index=False)

~/miniconda3/lib/python3.7/site-packages/featuretools/synthesis/dfs.py in dfs(entities, relationships, entityset, target_entity, cutoff_time, instance_ids, agg_primitives, trans_primitives, groupby_trans_primitives, allowed_paths, max_depth, ignore_entities, ignore_variables, seed_features, drop_contains, drop_exact, where_primitives, max_features, cutoff_time_in_index, save_progress, features_only, training_window, approximate, chunk_size, n_jobs, dask_kwargs, verbose, return_variable_types)
    220                                                   n_jobs=n_jobs,
    221                                                   dask_kwargs=dask_kwargs,
--> 222                                                   verbose=verbose)
    223     else:
    224         feature_matrix = calculate_feature_matrix(features,

~/miniconda3/lib/python3.7/site-packages/featuretools/computational_backends/calculate_feature_matrix.py in calculate_feature_matrix(features, entityset, cutoff_time, instance_ids, entities, relationships, cutoff_time_in_index, training_window, approximate, save_progress, verbose, chunk_size, n_jobs, dask_kwargs, profile)
    265                                                  cutoff_df_time_var=cutoff_df_time_var,
    266                                                  target_time=target_time,
--> 267                                                  pass_columns=pass_columns)
    268 
    269     feature_matrix = pd.concat(feature_matrix)

~/miniconda3/lib/python3.7/site-packages/featuretools/computational_backends/calculate_feature_matrix.py in linear_calculate_chunks(chunks, features, approximate, training_window, profile, verbose, save_progress, entityset, no_unapproximated_aggs, cutoff_df_time_var, target_time, pass_columns)
    528                                           cutoff_df_time_var,
    529                                           target_time, pass_columns,
--> 530                                           backend=backend)
    531         feature_matrix.append(_feature_matrix)
    532         # Do a manual garbage collection in case objects from calculate_chunk

~/miniconda3/lib/python3.7/site-packages/featuretools/computational_backends/calculate_feature_matrix.py in calculate_chunk(chunk, features, approximate, training_window, profile, verbose, save_progress, no_unapproximated_aggs, cutoff_df_time_var, target_time, pass_columns, backend, entityset)
    374                                                  cutoff_df_time_var: 'time'},
    375                                         inplace=True)
--> 376                     pass_through.set_index([id_name, 'time'], inplace=True)
    377                     for col in pass_columns:
    378                         _feature_matrix[col] = pass_through[col]

~/miniconda3/lib/python3.7/site-packages/pandas/core/frame.py in set_index(self, keys, drop, append, inplace, verify_integrity)
   4182             arrays.append(level)
   4183 
-> 4184         index = ensure_index_from_sequences(arrays, names)
   4185 
   4186         if verify_integrity and not index.is_unique:

~/miniconda3/lib/python3.7/site-packages/pandas/core/indexes/base.py in ensure_index_from_sequences(sequences, names)
   5313         return Index(sequences[0], name=names)
   5314     else:
-> 5315         return MultiIndex.from_arrays(sequences, names=names)
   5316 
   5317 

~/miniconda3/lib/python3.7/site-packages/pandas/core/indexes/multi.py in from_arrays(cls, arrays, sortorder, names)
    338         from pandas.core.arrays.categorical import _factorize_from_iterables
    339 
--> 340         codes, levels = _factorize_from_iterables(arrays)
    341         if names is None:
    342             names = [getattr(arr, "name", None) for arr in arrays]

~/miniconda3/lib/python3.7/site-packages/pandas/core/arrays/categorical.py in _factorize_from_iterables(iterables)
   2706         # For consistency, it should return a list of 2 lists.
   2707         return [[], []]
-> 2708     return map(list, lzip(*[_factorize_from_iterable(it) for it in iterables]))

~/miniconda3/lib/python3.7/site-packages/pandas/core/arrays/categorical.py in <listcomp>(.0)
   2706         # For consistency, it should return a list of 2 lists.
   2707         return [[], []]
-> 2708     return map(list, lzip(*[_factorize_from_iterable(it) for it in iterables]))

~/miniconda3/lib/python3.7/site-packages/pandas/core/arrays/categorical.py in _factorize_from_iterable(values)
   2678         # but only the resulting categories, the order of which is independent
   2679         # from ordered. Set ordered to False as default. See GH #15457
-> 2680         cat = Categorical(values, ordered=False)
   2681         categories = cat.categories
   2682         codes = cat.codes

~/miniconda3/lib/python3.7/site-packages/pandas/core/arrays/categorical.py in __init__(self, values, categories, ordered, dtype, fastpath)
    368 
    369                 # FIXME
--> 370                 raise NotImplementedError("> 1 ndim Categorical are not "
    371                                           "supported at this time")
    372 

NotImplementedError: > 1 ndim Categorical are not supported at this time
```
Because the cutoff_time dataframe's columns was in an order that was not expected. Once I did the following the problem was solved:

`ct = ct[["user_id", "time", "label"]]

I think this is not the intended behaviour as the cutoff_time parameter should not be that sensitive to the order of columns.

**Comment by kmax12** [comment]
@sebrahimi1988 that suggestion makes sense. would you open to trying to put up a PR for it?

**Comment by sebrahimi1988** [comment]
@kmax12 Sure, I'll have a look at the code!

**Comment by sebrahimi1988** [comment]
@kmax12 Could you add me as a contributor to the repo so I can open a PR?

**Comment by kmax12** [comment]
@sebrahimi1988 you should have to be added as contributor to open a PR. if you fork the repo, you will be able to submit a PR.

**Comment by thehomebrewnerd** [comment]
Closed by #969
